core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java

```diff
@@ -513,7 +513,7 @@ ResultHandle doLoad(MethodContext creator, MethodCreator method, ResultHandle ar
                 holder = substitutions.get(expectedType);
             }
             try {
-                ObjectSubstitution substitution = holder.sub.newInstance();
+                ObjectSubstitution substitution = holder.sub.getConstructor().newInstance();
```

Comment:
From https://docs.oracle.com/javase/9/docs/api/java/lang/Class.html#newInstance--: 

```suggestion
                ObjectSubstitution substitution = holder.sub.getDeclaredConstructor().newInstance();
```